apps/browser/qml/pages/components/BookmarkItem.qml

```diff
@@ -101,6 +94,17 @@ ListItem {
                 text: qsTrId("sailfish_browser-me-copy-to-clipboard")
                 onClicked: Clipboard.text = model.url
             }
+            MenuItem {
+                text: qsTrId("sailfish_browser-me-add_to_launcher")
+                onClicked: pageStack.animatorPush("AddToAppGridDialog.qml",
+                                                  {
+                                                      "url": url,
+                                                      "title": title,
+                                                      "icon": favicon,
+                                                      "desktopBookmarkWriter": desktopBookmarkWriter,
+                                                      "view": listView
```

Comment:
```suggestion
                                                      "bookmarkWriterParent": pageStack
```

There's potential object life-cycle problem now. The case where you have ""view": favoriteGrid" is fine from life-cycle point of view as the favoriteGrid is always there. However, in this case the saving will be aborted if you pop the Bookmark page before saving has finished.

Hmm, have to correct a bit... DesktopBookmarkWriter will block if there pending save upon destruction. Even so it be better not to block. Hence, ```bookmarkWriterParent: pageStack```.